sale_timesheet_task_exclude/tests/test_sale_timesheet_exclude_task.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests import common
+
+
+class TestSaleTimesheetExcludeTask(common.TransactionCase):
+
+    def setUp(self):
+        super().setUp()
+
+        self.uom_hour = self.env.ref('uom.product_uom_hour')
+        self.user_type_payable = self.env.ref(
+            'account.data_account_type_payable'
+        )
+        self.user_type_receivable = self.env.ref(
+            'account.data_account_type_receivable'
+        )
+        self.user_type_revenue = self.env.ref(
+            'account.data_account_type_revenue'
+        )
+        self.Partner = self.env['res.partner']
+        self.SudoPartner = self.Partner.sudo()
+        self.Employee = self.env['hr.employee']
+        self.SudoEmployee = self.Employee.sudo()
+        self.AccountAccount = self.env['account.account']
+        self.SudoAccountAccount = self.AccountAccount.sudo()
+        self.Project = self.env['project.project']
+        self.SudoProject = self.Project.sudo()
+        self.ProjectTask = self.env['project.task']
+        self.SudoProjectTask = self.ProjectTask.sudo()
+        self.AccountAnalyticLine = self.env['account.analytic.line']
+        self.SudoAccountAnalyticLine = self.AccountAnalyticLine.sudo()
+        self.ProductProduct = self.env['product.product']
+        self.SudoProductProduct = self.ProductProduct.sudo()
+        self.SaleOrder = self.env['sale.order']
+        self.SudoSaleOrder = self.SaleOrder.sudo()
+        self.SaleOrderLine = self.env['sale.order.line']
+        self.SudoSaleOrderLine = self.SaleOrderLine.sudo()
+        self.ProjectCreateSaleOrder = self.env['project.create.sale.order']
+
+    def test_1(self):
```

Comment:
This is not a good name, please think one that reflects what you are trying to prove

Reply:
Offer a name :)

Reply:
`test_exclude_from_sale_order` ?

Reply:
Sounds good to me, changed